src/Microsoft.Sbom.Api/Utils/SPDXFormatDetector.cs

```diff
@@ -17,17 +17,38 @@ public class SPDXFormatDetector : ISPDXFormatDetector
 {
     private readonly IFileSystemUtils fileSystemUtils;
     private readonly IDictionary<ManifestInfo, Func<Stream, bool>> supportedManifestInfos;
+    private readonly ISbomConfigFactory sbomConfigFactory;
 
     public SPDXFormatDetector(
         IFileSystemUtils fileSystemUtils,
-        IManifestParserProvider manifestParserProvider)
+        IManifestParserProvider manifestParserProvider,
+        ISbomConfigFactory sbomConfigFactory)
     {
         this.fileSystemUtils = fileSystemUtils;
         supportedManifestInfos = new Dictionary<ManifestInfo, Func<Stream, bool>>()
         {
             { Constants.SPDX22ManifestInfo, TryParse22 },
             { Constants.SPDX30ManifestInfo, TryParse30 }
         };
+        this.sbomConfigFactory = sbomConfigFactory;
+    }
+
+    public bool TryGetSbomsWithVersion(string manifestDirPath, out IDictionary<string, ManifestInfo> detectedSboms)
+    {
+        detectedSboms = new Dictionary<string, ManifestInfo>();
+        foreach (var mi in supportedManifestInfos.Keys) {
+            var filePath = sbomConfigFactory.GetSbomFilePath(manifestDirPath, mi);
+            if (fileSystemUtils.FileExists(filePath))
+            {
+                var result = TryDetectFormat(filePath, out var detectedManifestInfo);
+                if (result && mi.Equals(detectedManifestInfo))
```

Comment:
@DaveTryon do you have any opinion on the behavior in the (very edge) case where a SBOM with spdx2.2 in the path is actually a 3.0 SBOM (or vice versa)? This now this logic would just ignore it. Do you even think it's worthwhile to attempt to parse a SBOM with 2.2 in the path as 3.0 and vice versa?

Reply:
Discussed offline and we will mention this to John Wade to get his input on this behavior